### PR TITLE
SNOW-584369 get_results_from_sfqid missing errno

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-584369

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Errors raised when running `get_results_from_sfqid` are missing `errno` numbers. In addition I found that we had no nice error messages when a query is not found by the same method. So I added error checking and nicer error message when this happens.

### CODE REVIEW

**Review by sfc-gh-jdu — APPROVED** [review]

File: src/snowflake/connector/connection.py [hunk 1]
```diff
@@ -1433,19 +1434,30 @@ def get_query_status_throw_if_error(self, sf_qid: str) -> QueryStatus:
         """
         status, status_resp = self._get_query_status(sf_qid)
         queries = status_resp["data"]["queries"]
+        try:
+            found_query = queries[0]
+        except IndexError:
+            Error.errorhandler_wrapper(
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/connection.py:1440]: would this be a breaking change to user?
we wrap an `IndexError` into a `ProgrammingError`, user code catching `IndexError` previously won't work now.

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/connection.py:1440]: Yes, I was okay waiting for this to be pre-announced, but maybe I should just split this PR into 2

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/connection.py:1440]: Split into #1159

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/connector/connection.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/connector/connection.py]: this change looks totally reasonable to me.
However, errors are part of the public apis, I'm just curious that when we make public api changes, is there any procedure (e.g. approval) we need to go through or we just make the change and announce it.

File: DESCRIPTION.md [hunk 2]
```diff
@@ -10,13 +10,18 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 
 
+- v2.7.9()
```
Comment (sfc-gh-yixie) [inline DESCRIPTION.md:13]: Suggest `v2.7.9(Unreleased)`

File: src/snowflake/connector/connection.py (hunk 1 above)
Comment (sfc-gh-yixie) [reply src/snowflake/connector/connection.py]: @sfc-gh-mkeller What do we usually do for bug-fixing breaking change?

**Review by sfc-gh-yixie — APPROVED** [review]

File: src/snowflake/connector/connection.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/connector/connection.py]: thanks for the info!

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**